From 826a630149db3006cb1f6ddabacbcb738b21c3bc Mon Sep 17 00:00:00 2001
From: Bowgo Tsai <bowgotsai@google.com>
Date: Sat, 26 Jan 2019 17:14:03 +0900
Subject: [PATCH 1/1] Adding better AVB error message

Also removing the requirement of BOARD_BOOTIMAGE_PARTITION_SIZE when
BOARD_AVB_ENABLE is set. Some targets (e.g., emulator) doesn't build
boot.img when AVB is enabled.

Bug: 112293933
Test: build a failure case and checks the error output is expected
Change-Id: I290c707719193ddaedcd9fd0b31de566ad17078c
Merged-In: I290c707719193ddaedcd9fd0b31de566ad17078c
(cherry picked from commit 99ed1b4fbd5124f344a95a05a1f69e9a93b9d0dd)
Signed-off-by: AndroPlus <mail@andro.plus>
---
 core/Makefile                     |  4 ----
 tools/releasetools/build_image.py | 10 +++++++---
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index 9f9a9f0c8..10fc98479 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -2373,10 +2373,6 @@ ifdef BOARD_AVB_ROLLBACK_INDEX
 BOARD_AVB_MAKE_VBMETA_IMAGE_ARGS += --rollback_index $(BOARD_AVB_ROLLBACK_INDEX)
 endif
 
-ifndef BOARD_BOOTIMAGE_PARTITION_SIZE
-  $(error BOARD_BOOTIMAGE_PARTITION_SIZE must be set for BOARD_AVB_ENABLE)
-endif
-
 # $(1): the directory to extract public keys to
 define extract-avb-chain-public-keys
   $(if $(BOARD_AVB_BOOT_KEY_PATH),\
diff --git a/tools/releasetools/build_image.py b/tools/releasetools/build_image.py
index 5a7fe9e26..a97a75cf0 100755
--- a/tools/releasetools/build_image.py
+++ b/tools/releasetools/build_image.py
@@ -198,8 +198,11 @@ def AVBAddFooter(image_path, avbtool, footer_type, partition_size,
 
   cmd.extend(shlex.split(additional_args))
 
-  (_, exit_code) = RunCommand(cmd)
-  return exit_code == 0
+  output, exit_code = RunCommand(cmd)
+  if exit_code != 0:
+    print("Failed to add AVB footer! Error: %s" % output)
+    return False
+  return True
 
 
 def AdjustPartitionSizeForVerity(partition_size, fec_supported):
@@ -558,7 +561,8 @@ def BuildImage(in_dir, prop_dict, out_file, target_out=None):
     additional_args = prop_dict["avb_add_" + avb_footer_type + "_footer_args"]
     max_image_size = AVBCalcMaxImageSize(avbtool, avb_footer_type,
                                          partition_size, additional_args)
-    if max_image_size == 0:
+    if max_image_size <= 0:
+      print("AVBCalcMaxImageSize is <= 0: %d" % max_image_size)
       return False
     prop_dict["partition_size"] = str(max_image_size)
     prop_dict["original_partition_size"] = partition_size
-- 
2.17.1

